### PR TITLE
Add all missing dependencies for Setup project to be able to compile it from Visual Studio

### LINKED ISSUE #2745: Make Setup project compilable straight from Visual Studio
Discussing #2729 I found the following problems
1. Setup project relies on the existence of bin\Release folders that don't contain the desired files if you build in Debug configuration. So at least configuration project should not be compiling in Debug configuration

Setup projects relies on the existence of external installers. The system cannot find the file '..\Bin\KDiff3-64bit-Setup_0.9.98-2.exe' and those are downloaded by **DownloadExternals.cmd** which should be added as a prebuild step to the Setup project

### PR DESCRIPTION
**Pull request by mnaoumov**
Fixes #2745

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -59,3 +59,5 @@ UpgradeLog*.htm*
 packages/
 GitExtensions.*.sln.VisualState.xml
 GitExtensions.settings.backup
+Bin/Git-1.9.5-preview20141217.exe
```
Comment (Radrik5) [inline .gitignore]: Please consider Bin/Git-* pattern that includes future versions as well.
The same for KDiff3.

Comment (mnaoumov) [inline .gitignore]: Yeah, I thought so but then I scared to have too broad filter. Now I am not scared anymore and put this in ;)

**Comment by sharwell** [comment]
I would prefer to see someone create a NuGet package containing these binaries, and add them to the solution-level **packages.config**.

**Comment by RussKie** [comment]
@mnaoumov I know it has been a long while... 
Do you still wish to get this in?

File: Setup/Setup.wixproj [hunk 2]
```diff
@@ -66,6 +66,30 @@
     <Folder Include="UI" />
   </ItemGroup>
   <Import Project="$(WixTargetsPath)" />
+  <Target Name="BeforeBuild">
```
Comment (sharwell) [inline Setup/Setup.wixproj]: ❗️ This is not a particularly reliable way of accomplishing this.

**Comment by sharwell** [comment]
😆 I just saw this on my notifications page. I totally didn't realize that I had previously commented.

**Comment by RussKie** [comment]
If you can help make it better - please do so :)

**Comment by sharwell** [comment]
@RussKie Sure I think I have a little bit of time today. I will take a look.

**Pull request closed without merging**